### PR TITLE
Adds documentation

### PR DESCRIPTION
**Pull request by meiamsome**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+class IntrospectedGraphQLService extends GraphQLAutoRequesterDataSource {
+  constructor() {
+    this.baseURL = 'https://example.com/'
+  }
+}
+```
+
+###  Using A Schema Document
+
+If you wish to use a Schema Document, you must set the `baseURL` for your service in the constructor, and provide the
```
Comment (animalcharade) [inline README.md:22]:
```suggestion
If you wish to use a Schema Document, you must set the `baseURL` for your service in the constructor and provide the
```

Comment (animalcharade) [reply README.md:22]: I think there's an argument to be made for retaining this comma, but I personally feel that it works better without.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+schema document as a property on the class itself.
```
Comment (animalcharade) [inline README.md:23]:
```suggestion
schema document as a property on the class itself:
```

Comment (animalcharade) [reply README.md:23]: You previously used a colon at the end of your other "If you wish to..." statement.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+SchemaDocumentGraphQLService.schemaDocument = `
+  type Query {
+    example: Int!
+  }
+`
+```
+
+###  Using An executable Schema
+
+If you wish to use an executable Schema, you can set the `executableSchema` property in the constructor
```
Comment (animalcharade) [inline README.md:40]:
```suggestion
If you wish to use an executable Schema, you can set the `executableSchema` property in the constructor:
```

File: README.md (hunk 1 above)
Comment (steve7411) [inline README.md:22]: I don't believe 'and' is functioning as a conjunction here

File: README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  constructor() {
+    this.executableSchema = myExecutableSchema
+  }
+}
+```
+
+## Usage as a DataSource:
+
+Once you have your dataSource configured in your Apollo Server initialization, you can then access the `.query` field
+from the dataSources key in your resolvers to start a query against the upstream service. For more information on the
```
Comment (animalcharade) [inline README.md:52]: You previously used DataSources on line 7, with a capital D, but you're using a lowercase D here. Intentional?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+usage of the `graphql-auto-requester` see [its
+documentation](https://github.com/meiamsome/graphql-auto-requester/blob/master/Readme.md)
```
Comment (animalcharade) [inline README.md:54]:
```suggestion
documentation](https://github.com/meiamsome/graphql-auto-requester/blob/master/Readme.md).
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+class ExecutableSchemaGraphQLService extends GraphQLAutoRequesterDataSource {
+  constructor() {
+    this.executableSchema = myExecutableSchema
+  }
+}
+```
+
+## Usage as a DataSource:
+
+Once you have your dataSource configured in your Apollo Server initialization, you can then access the `.query` field
```
Comment (steve7411) [inline README.md:51]: field?

**Pull request merged**